draft-specification.md

```diff
@@ -0,0 +1,251 @@
+# Draft Syrup Specification
+
+## Status of This Document
+
+This is a draft specification, created by the [OCapN](https://github.com/ocapn/ocapn) pre-standardization group. This draft is subject to change with the work of the group; if you're interested in being part of that work, please join!
+
+## Overview
+
+Syrup is a lightweight and easy-to-implement data serialization format. A serialized Syrup structure is canonicalized; the same data thus always serializes to the same set of octets (8 bit values).
+
+Syrup is a *binary* serialization format; data is not encoded within text, but as the raw underlying data, preventing common escaping bugs. Syrup also provides an easy way to canonicalized as unordered data types are sorted by the octet which is simple to implement.
```

Comment:
I see you disagree with it, but it's simpler to do from an *implementation* standpoint, even if from a conceptual/CS standpoint, it's a bit silly potentially.  It's trivially easy to implement in a recursive descent writer.  That's part of the goal of Syrup, even if it's a bit silly.  Nice from an engineering perspective, dissatisfying from a CS perspective.

(I understand that ASN.1 did the same thing, and was one of the few parts of that ecosystem that maybe wasn't so complicated.)

Reply:
I don't see how it's simpler from an implementation standpoint, and I don't think you've understood my critique, since the bulk of my complaint was that the opposite was true (in addition to being inefficient).

...frankly though, it seems like the data model we're converging on in e.g. #5 kindof obviates this; if we're roughly agreed upon things like no maps & sets, only strings as labels for Tagged, and other changes that are clearly incompatible with preserves, then does it even make sense for us to merge a spec for syrup at this point? It seems clear we won't use it in the end.